src/Writers/HTMLWriter.jl

```diff
@@ -1218,16 +1218,23 @@ end
 
 function domify(ctx, navnode, anchor::Anchors.Anchor)
     @tags a
-    aid = "$(anchor.id)-$(anchor.nth)"
+    frag = Anchors.fragment(anchor)
+    legacy = if anchor.nth == 1
+        (a[:id => lstrip(frag, '#')*"-1"],)
+    else
+        ()
+    end
     if isa(anchor.object, Markdown.Header)
         h = anchor.object
         fixlinks!(ctx, navnode, h)
-        DOM.Tag(Symbol("h$(Utilities.header_level(h))"))[:id => aid](
-            a[".docs-heading-anchor", :href => "#$aid"](mdconvert(h.text, h)),
-            a[".docs-heading-anchor-permalink", :href => "#$aid", :title => "Permalink"]
+        DOM.Tag(Symbol("h$(Utilities.header_level(h))"))[:id => lstrip(frag, '#')](
+            a[".docs-heading-anchor", :href => frag](mdconvert(h.text, h)),
+            legacy...,
+            a[".docs-heading-anchor-permalink", :href => frag, :title => "Permalink"]
         )
     else
-        a[:id => aid, :href => "#$aid"](domify(ctx, navnode, anchor.object))
+        error("hmmm...")
```

Comment:
This path seems to not be used in Documenter at all. Can you even attach `id` to things that are not headers? I guess we can keep it, but maybe no need to add the legacy link here.